model/patch_lifecycle.go

```diff
@@ -149,9 +149,18 @@ func GetPatchedProject(ctx context.Context, p *patch.Patch, githubOauthToken str
 	}
 	opts.RemotePath = path
 	opts.PatchOpts.env = env
-	projectFileBytes, err = getFileForPatchDiff(ctx, *opts)
-	if err != nil {
-		return nil, "", errors.Wrapf(err, "could not fetch remote configuration file")
+
+	if githubOauthToken == evergreen.Localhost {
+		opts.ReadFileFrom = ReadFromLocal
+		projectFileBytes, err = ioutil.ReadFile(path)
```

Comment:
If we require the config file to be changed to be able to patch locally, do we hit https://github.com/evergreen-ci/evergreen/blob/17b6f385ea0f76f5173772c66c303b14ebc06ea4/model/project_parser.go#L745 instead? That would be preferable to modifying this code just to handle the local case.

Reply:
i looked into this for a while but configChanged doesnt return the correct result (i think) because module name always seems to be populated even at the creation of patchDoc. im not sure what would be a better alternative

Reply:
Hmm that shouldn't be true -- one of them will always be the empty string because it's the current repo. Can we add some logging to ConfigChanged to walk through this together today?